xma_data_processor.py

```diff
@@ -0,0 +1,374 @@
+"""Converts data from XMALab into the format useful for DLC training."""
+
+import os
+
+import blend_modes
+import cv2
+import numpy as np
+import pandas as pd
+from ruamel.yaml import YAML
+
+
+class XMADataProcessor:
+    """Converts data from XMALab into the format useful for DLC training."""
+
+    def __init__(self, config):
+        self._config = config
+        self._mode = config["tracking_mode"]
+
+    def get_bodyparts_from_xma(
+        self, path_to_trial, split_markers=False, crossed_markers=False
+    ):
+        """Pulls the names of the XMAlab markers from the 2Dpoints file"""
+
+        csv_path = [file for file in os.listdir(path_to_trial) if file[-4:] == ".csv"]
+        if len(csv_path) > 1:
+            raise FileExistsError(
+                "Found more than 1 CSV file for trial: " + path_to_trial
+            )
+        if len(csv_path) <= 0:
+            raise FileNotFoundError(
+                "Couldn't find a CSV file for trial: " + path_to_trial
+            )
+        trial_csv = pd.read_csv(
+            path_to_trial + "/" + csv_path[0],
+            sep=",",
+            header=0,
+            dtype="float",
+            na_values="NaN",
+        )
+        names = trial_csv.columns.values
+        if self._mode == "rgb":
+            parts = [name.rsplit("_", 1)[0] for name in names]
+            if split_markers:
```

Comment:
Split and crossed markers should also be loadable from the config... something to look into.

Reply:
(to clarify, since I was confused reading my own comment here, there are booleans for both split and crossed markers that should be loaded in with the config (and therefore accessible from the object without any references/function parameters).

Reply:
done, removed function parameter here & in _splice_xma_to_dlc